cmake/load_openmp.cmake

```diff
@@ -8,3 +8,9 @@ else ()
   message(STATUS "Did not find OpenMP")
   set(VT_TV_OPENMP_ENABLED OFF)
 endif()
+
+function(link_openmp)
+  if(VT_TV_OPENMP_ENABLED)
+    target_link_libraries(${VT_TV_LIBRARY} PUBLIC OpenMP::OpenMP_CXX)
```

Comment:
We should be using per target settings instead of modifying `CMAKE_CXX_FLAGS` and such. I will create a follow-up issue to track this.